static/js/containers/ProfilePage.js

```diff
@@ -2,105 +2,95 @@
 import ProfileFormContainer from './ProfileFormContainer';
-import PersonalTab from '../components/PersonalTab';
-import EmploymentTab from '../components/EmploymentTab';
-import EducationTab from '../components/EducationTab';
+import { validateProfileComplete } from '../lib/validation/profile';
+import type { Profile, ProfileGetResult } from '../flow/profileTypes';
 import {
-  PERSONAL_STEP,
-  EDUCATION_STEP,
-  EMPLOYMENT_STEP,
-} from '../constants';
-import { createActionHelper } from '../lib/redux';
-import type { Profile } from '../flow/profileTypes';
-import { startProfileEdit } from '../actions/profile';
+  makeProfileProgressDisplay,
+  currentOrFirstIncompleteStep,
+} from '../util/util';
 
 class ProfilePage extends ProfileFormContainer {
-  currentStep: Function = (): string => {
-    const { ui: { profileStep } } = this.props;
-    return profileStep;
-  };
-
   componentDidMount() {
     const { dispatch } = this.props;
     dispatch(startProfileEdit(SETTINGS.user.username));
   }
 
-  stepTransitions: Function = (): [void|() => void, () => void] => {
-    const { dispatch } = this.props;
-    let setStep = createActionHelper(dispatch, setProfileStep);
-    let createStepFunc = step => () => {
-      setStep(step);
-      ReactDOM.findDOMNode(this).querySelector(".profile-pagination").scrollIntoView();
-    };
-    switch ( this.currentStep() ) {
-    case EDUCATION_STEP:
-      return [createStepFunc(PERSONAL_STEP), createStepFunc(EMPLOYMENT_STEP)];
-    case EMPLOYMENT_STEP:
-      return [
-        createStepFunc(EDUCATION_STEP),
-        () => this.context.router.push('/dashboard')
-      ];
-    default:
-      return [undefined, createStepFunc(EDUCATION_STEP)];
-    }
-  };
+  componentDidUpdate() {
+    const username = SETTINGS.user.username;
+    const {
+      profiles: {
+        [username]: profileInfo,
+      },
+      ui: { profileStep },
+    } = this.props;
+
+    // check that this is a valid step to be on
+    if (profileStep !== null &&
+        profileInfo !== undefined &&
+        profileInfo.getStatus !== FETCH_PROCESSING) {
+      const { profile } = this.profileProps(profileInfo);
 
-  currentComponent: Function = (props): React$Element<*> => {
-    switch ( this.currentStep() ) {
-    case EDUCATION_STEP:
-      return <EducationTab {...props} />;
-    case EMPLOYMENT_STEP:
-      return <EmploymentTab {...props} />;
-    default:
-      return <PersonalTab {...props} />;
+      if (profile !== undefined) {
+        const [ , step, ] = validateProfileComplete(profile);
+        // determine the ideal step we should be on
+        // this will either be the current step if all preceding steps are complete
+        // or the least complete step below the current one
+        let idealStep = currentOrFirstIncompleteStep(profileStep, step);
+        if (profileStep !== idealStep) {
+          // if learner is on a step they shouldn't be on yet, redirect them to the correct one
```

Comment:
I don't think these comments are super necessary, you can keep them if you want though.